Workbooks/IoT Central/IoT Edge device details/iotedge-device-details.workbook

```diff
@@ -0,0 +1,3347 @@
+        "value": "not set"
+      },
+      "name": "parameters - 5"
+    },
+    {
+      "type": 1,
+      "content": {
+        "json": "# No metrics data from IoT Edge devices detected in the last 3 days\n\n## Please on-board with instructions at **[aka.ms/edgemon-docs](https://aka.ms/edgemon-docs)**\n\n<br>\n💡*If you've just on-boarded,  it can take up to 15 minutes for data to be displayed.*"
+      },
+      "conditionalVisibility": {
+        "parameterName": "InsightsMetricsExists",
+        "comparison": "isEqualTo",
+        "value": "no"
+      },
+      "name": "text - 4"
+    },
+    {
+      "type": 9,
+      "content": {
+        "version": "KqlParameterItem/1.0",
+        "parameters": [
+          {
+            "id": "1b4b2d2e-4f2e-4881-903d-f307d6958a1c",
+            "version": "KqlParameterItem/1.0",
+            "name": "timeRange",
+            "label": "Time range",
+            "type": 4,
+            "isRequired": true,
+            "isGlobal": true,
+            "value": {
+              "durationMs": 14400000
+            },
+            "typeSettings": {
+              "selectableValues": [
+                {
+                  "durationMs": 3600000
+                },
+                {
+                  "durationMs": 14400000
+                },
+                {
+                  "durationMs": 43200000
+                },
+                {
+                  "durationMs": 86400000
+                },
+                {
+                  "durationMs": 172800000
+                },
+                {
+                  "durationMs": 259200000
+                },
+                {
+                  "durationMs": 604800000
+                },
+                {
+                  "durationMs": 1209600000
+                },
+                {
+                  "durationMs": 2592000000
+                }
+              ],
+              "allowCustom": true
+            }
+          }
+        ],
+        "style": "pills",
+        "queryType": 0,
+        "resourceType": "{resourceType}"
+      },
+      "conditionalVisibility": {
+        "parameterName": "InsightsMetricsExists",
+        "comparison": "isEqualTo",
+        "value": "yes"
+      },
+      "customWidth": "0",
+      "name": "parameters - 1 - Copy"
+    },
+    {
+      "type": 9,
+      "content": {
+        "version": "KqlParameterItem/1.0",
+        "parameters": [
+          {
+            "id": "63aa6750-da91-4e11-af98-50e156843464",
+            "version": "KqlParameterItem/1.0",
+            "name": "SelectedDevice",
+            "label": "💻 Choose device",
+            "type": 2,
+            "query": "\nlet startTime = {timeRange:start};\nlet endTime = {timeRange:end};\nlet devices = () { \nInsightsMetrics\n| where Name == \"edgehub_gettwin_total\"\n| where TimeGenerated > startTime and TimeGenerated < endTime\n| extend dimensions=parse_json(Tags)\n| extend device = tostring(dimensions.edge_device)\n| distinct device\n};\ndevices()",
```

Comment:
FYI, for the time part, you don't need to use let to get start/end.  there's also a default syntax that will be generated if you do

```
where TimeGenerated {timeRange}
```

which will generate 

```
where TimeGenerated between( "something" .. "something else" )
``

unless you need start/end, this syntax can generally make for much simpler queries

The real reason i commented here though is that this query fails on an iot app that isn't onboarded:

![image](https://user-images.githubusercontent.com/10158007/125985257-6308d732-cada-4507-ab61-131708e0fa77.png)


you're already hiding/showing other things if not onboarded, not sure if this was intentional or missed here?